### ISSUE TITLE
Exception when used with SQLAlchemy 0.9.4

### ISSUE THREAD

**Issue by erikcw** [aldjemy/aldjemy#15]
I'm trying to use aldjemy with SQLAlchemy 0.9.4.  When I try a simple query, I get this traceback:

``` python

from django.contrib.auth.models import User
User.sa.query(User.sa).count()

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-01a7a3ebfca2> in <module>()
----> 1 User.sa.query(User.sa).count()

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/orm/query.pyc in count(self)
   2591         """
   2592         col = sql.func.count(sql.literal_column('*'))
-> 2593         return self.from_self(col).scalar()
   2594 
   2595     def delete(self, synchronize_session='evaluate'):

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/orm/query.pyc in scalar(self)
   2390         """
   2391         try:
-> 2392             ret = self.one()
   2393             if not isinstance(ret, tuple):
   2394                 return ret

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/orm/query.pyc in one(self)
   2359 
   2360         """
-> 2361         ret = list(self)
   2362 
   2363         l = len(ret)

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/orm/loading.pyc in instances(query, cursor, context)
     63                 break
     64         else:
---> 65             fetch = cursor.fetchall()
     66 
     67         if custom_rows:

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/engine/result.pyc in fetchall(self)
    786             self.connection._handle_dbapi_exception(
    787                                     e, None, None,
--> 788                                     self.cursor, self.context)
    789 
    790     def fetchmany(self, size=None):

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/engine/base.pyc in _handle_dbapi_exception(self, e, statement, parameters, cursor, context)
   1109                                 )
   1110 
-> 1111             util.reraise(*exc_info)
   1112 
   1113         finally:

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/engine/result.pyc in fetchall(self)
    781         try:
    782             l = self.process_rows(self._fetchall_impl())
--> 783             self.close()
    784             return l
    785         except Exception as e:

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/engine/result.pyc in close(self, _autoclose_connection)
    541             if _autoclose_connection and \
    542                 self.connection.should_close_with_result:
--> 543                 self.connection.close()
    544             # allow consistent errors
    545             self.cursor = None

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/engine/base.pyc in close(self)
    635         else:
    636             if not self.__branch:
--> 637                 conn.close()
    638             if conn._reset_agent is self.__transaction:
    639                 conn._reset_agent = None

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/pool.pyc in close(self)
    781         self._counter -= 1
    782         if self._counter == 0:
--> 783             self._checkin()
    784 
    785 

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/pool.pyc in _checkin(self)
    664     def _checkin(self):
    665         _finalize_fairy(self.connection, self._connection_record,
--> 666                             self._pool, None, self._echo, fairy=self)
    667         self.connection = None
    668         self._connection_record = None

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/pool.pyc in _finalize_fairy(connection, connection_record, pool, ref, echo, fairy)
    564 
    565     if connection_record:
--> 566         connection_record.checkin()
    567 
    568 

/Users/erik/.virtualenvs/proj/lib/python2.7/site-packages/sqlalchemy/pool.pyc in checkin(self)
    455         connection = self.connection
    456         pool = self.__pool
--> 457         while self.finalize_callback:
    458             finalizer = self.finalize_callback.pop()
    459             finalizer(connection)

AttributeError: '_ConnectionRecord' object has no attribute 'finalize_callback'
```

Same exception regardless of the type of query I attempt.  What is needed to make aldjemy compatible with the current release of SQLAlchemy?

**Comment by ojomio** [comment]
The issue re-occurs with SQLAlchemy 1.0.5 and aldjemy 0.4.1

**Comment by erikcw** [comment]
@ojomio Can you post your full traceback?

**Comment by ojomio** [comment]
@erikcw , err, I'm afraid the most relevant part of it is 

'_ConnectionRecord' object has no attribute 'starttime'
```
C:\Python34\lib\site-packages\sqlalchemy\pool.py in _invalidate
        if not rec or self._invalidate_time < rec.starttime: ...
▼ Local vars
Variable	Value
rec	<aldjemy.core._ConnectionRecord object at 0x0735E730>
connection	<sqlalchemy.pool._ConnectionFairy object at 0x0735E770>
exception	InterfaceError('connection already closed',)
self	<aldjemy.core.DjangoPool object at 0x048915F0>
```

**Comment by ojomio** [comment]
@erikcw Actually, it seems with this match on sqlalchemy=1.0.6 it doesn't work at all... 

 It raises ```Interface Error: Connection already closed```, which in SqlAlchemy 0.9.{4,8} was expressed differently with absent attribute. I mean, the root is a _closed connection_ after all. Why this can ever happen?! 

With SA 0.9.{4,8} and Aldjemy=0.4.0 I manage to make like successful 4-5 requests to Django and then I get ```AttributeError```. With SA 1.0.6 and this ```master``` version of aldjemy I am bumped right away with ```InterfaceError```

**Comment by erikcw** [comment]
@ojomio Take a look at #32 -- I ran into the same issue.  Should probably document your findings on that ticket.
